sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -250,11 +278,33 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
   }
 
   private def listTestCases(): Seq[TestCase] = {
-    listFilesRecursively(new File(inputFilePath)).map { file =>
-      val resultFile = file.getAbsolutePath.replace(inputFilePath, goldenFilePath) + ".out"
-      val absPath = file.getAbsolutePath
-      val testCaseName = absPath.stripPrefix(inputFilePath).stripPrefix(File.separator)
-      TestCase(testCaseName, absPath, resultFile)
+    listFilesRecursively(new File(inputFilePath)).flatMap { file =>
+      testCases(file.getAbsolutePath)
+    }
+  }
+
+  private def testCases(inputPath: String): Seq[TestCase] = {
+    val baseResultFileName = inputPath.replace(inputFilePath, goldenFilePath)
+    val testCaseName = inputPath.stripPrefix(inputFilePath).stripPrefix(File.separator)
+    testConfigs.get(testCaseName) match {
```

Comment:
Once a test is renamed, it might silently turn to default test config. To avoid that, maybe we should explicitly define which test case needs to run against custom configs.

Reply:
Mmmh... Well, actually it is not very silently, as you would see it executes only once it the test runs and not many times with different suffixes. But of course it won't fail. @cloud-fan @maropu  what do you think?

Reply:
If we need to take care of the issue, how about listing up all the test files inside `SQLQueryTestSuite.scala`? Then, if developers add/delete/rename test files, they need to update the list?

Reply:
Honestly I do not really like this idea @maropu, it puts an extra effort which is not really needed...